### PR TITLE
Add environment notification messages

### PR DESCRIPTION
**Pull request by dsrivastavv**

### CODE REVIEW

File: lisa/environment.py [hunk 1]
```diff
@@ -206,9 +226,11 @@ def create(
         id_: int,
```
Comment (squirrelsc) [inline lisa/environment.py]: It looks we can remove the `create` method, to create new instance from `__init__` directly.

Comment (squirrelsc) [reply lisa/environment.py]: originally, there may be different code path to create env, but for new, the logic is pretty simple. Let's remove the class method `create`.

Comment (dsrivastavv) [reply lisa/environment.py]: My first thought was to remove this method as well as we discussed offline. However, it seems like nodes are being initialized in different ways. For example
https://github.com/microsoft/lisa/blob/5c72dabe3151d869eb15b827758f3c9bc959fe5b/lisa/environment.py#L240-L249

https://github.com/microsoft/lisa/blob/5c72dabe3151d869eb15b827758f3c9bc959fe5b/lisa/sut_orchestrator/azure/tests/test_prepare.py#L312-L318

The code in test_prepare.py isn't calling the code in `create`. Was `create` method written to make testing easier?

Comment (squirrelsc) [reply lisa/environment.py]: The test code looks creating the default Azure extension schema, so it should be ok to without the create method. It needs create runbook, and then pass it in.

**Comment by dsrivastavv** [comment]
@squirrelsc One of the test case is failing because it use node_req_count=0. We will either have to remove this testcase or the following assert check : 
`if not runbook.nodes_requirement and not runbook.nodes`  

I am leaning towards removing the testcase as assert would be more useful for users. Thoughts?

**Comment by squirrelsc** [comment]
> @squirrelsc One of the test case is failing because it use node_req_count=0. We will either have to remove this testcase or the following assert check :
> `if not runbook.nodes_requirement and not runbook.nodes`
> 
> I am leaning towards removing the testcase as assert would be more useful for users. Thoughts?

Sure, let's remove that test case.

File: lisa/environment.py [hunk 2]
```diff
@@ -146,34 +155,56 @@ def _expand_node_space(self) -> None:
...
+        self.runbook = runbook
+        self.name = runbook.name
         self.is_predefined: bool = is_predefined
         self.is_new: bool = True
         self.id = id_
+        self.warn_as_error = warn_as_error
         self.platform: Optional[Platform] = None
+        self._default_node: Optional[Node] = None
+        self._log = get_logger("env", self.name)
+        has_default_node = False
```
Comment (squirrelsc) [inline lisa/environment.py]: it's a local variable, put it near to its logic.

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**